File: Data Structures & Algorithms/longest-common-prefix/submission-2.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        x = 0
        while x < len(strs[0]):
            ch = strs[0][x]
            for i in range(1, len(strs)):
                if x == len(strs[i]) or ch != strs[i][x]:
                    return strs[0][0:x]

            x += 1

        return strs[0]
        

File: Data Structures & Algorithms/anagram-groups/submission-2.js
class Solution {
    /**
     * @param {string[]} strs
     * @return {string[][]}
     */
    groupAnagrams(strs) {
        let map = {};

        for(let i =0; i<strs.length; i++) {
            let freqArr = Array(26).fill(0);

            let str = strs[i];
            for(let j =0; j<str.length; j++) {
                let idx = str[j].charCodeAt() - 'a'.charCodeAt();
                ++freqArr[idx];
            }

            let key = '';
            for(let k =0; k<freqArr.length; k++) {
                key += "#" + freqArr[k];
            }
            
            if(!map[key]) {
                map[key] = [strs[i]];
            } else {
                map[key].push(strs[i]);
            }
        }
        return [...Object.values(map)];
    }
}
